app/src/main/java/org/oppia/app/home/continueplaying/ContinuePlayViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package org.oppia.app.home.continueplaying
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import org.oppia.app.R
+import org.oppia.app.model.OngoingStoryList
+import org.oppia.domain.topic.TopicListController
+import org.oppia.util.data.AsyncResult
+import javax.inject.Inject
+
+// TODO(#297): Add download status information to promoted-story-card.
+
+/** [ViewModel] for displaying a promoted story. */
+class ContinuePlayViewModel @Inject constructor(
+  private val fragment: Fragment,
+  private val topicListController: TopicListController
+) : ContinuePlayingItemViewModel() {
+
+  val itemList: MutableList<ContinuePlayingItemViewModel> = ArrayList()
+
+  val ongoingStoryListSummaryResultLiveData: LiveData<AsyncResult<OngoingStoryList>> by lazy {
+    topicListController.getOngoingStoryList()
+  }
+
+  val onGoingStoryLiveData: LiveData<List<ContinuePlayingItemViewModel>>by lazy {
+    Transformations.map(ongoingStoryListSummaryResultLiveData, ::processOngoingStoryList)
+  }
+
+  private fun processOngoingStoryList(ongoingStoryList: AsyncResult<OngoingStoryList>): List<ContinuePlayingItemViewModel> {
+    if (ongoingStoryList.isSuccess()) {
+      if (ongoingStoryList.getOrThrow().recentStoryCount > 0) {
```

Comment:
Better approach is to write it this way:
`if (ongoingStoryList.getOrThrow().recentStoryList.isNotEmpty()) {`

Reply:
Updated.